### DISCUSSION TITLE
[FullStack] Elixir + Phoenix

### DISCUSSION THREAD

**Discussion by fegaeze** [realworld-apps/realworld#858] [WIP Implementations]
Hello!!! Will be working on a full-stack realworld demo application using Elixir and Phoenix 👌🏾
[Link to repository](https://github.com/fegaeze/realworld-elixir-fullstack)

**Comment by pyrsmk** [comment]
Isn't that the third Phoenix demo ?

**Comment by geromegrignon** [comment]
Indeed We already include 2 related implementations and won't add a new one @fegaeze.
I'll take some time to review the gothinkster one as we have limited maintenance on it.

**Reply by fegaeze** [reply]
Ohk. Thanks

**Reply by mattneel** [reply]
To be fair none of those examples are actually full stack (frontend + backend), and none of them are recent enough to be using LiveView.

**Reply by benregn** [reply]
At the time of writing this, I can only find one example implementation for Phoenix, which is rather outdated (about to be two versions behind) and also "just" a backend API implementation and not a fullstack one.

As an example there are three (fullstack) implementations for *Laravel*. One is using Laravel as a more traditional API with a frontend framework (Vue.js), another is using Laravel with Livewire, which is Laravel's take on Phoenix's LiveView, and there is even one which is the same Laravel + Vue.js combo as an existing one but with TypeScript thrown in.

Another example is *Django*. From what I can see there appear to be two (fullstack) implementations for Django with "unobtrusive" Javascript frameworks. One is Django with Unpoly and the other is Django with HTMX and Alpine.js. So perhaps two rather similar (at least in ideas) implementations for the same framework.

Just from these examples for those two frameworks, it looks like there would be room for a different Phoenix implementation, as long as it shows off how to do it using LiveView. @geromegrignon Is there room to reconsider this stance?

**Comment by martinstannard** [comment]
I'd like to second the request to consider a Phoenix implementation using LiveView. LiveView brings a different programming model that is worthy of inclusion on its own, and given there is already a Laravel version using LiveWire it seems that there is scope for inclusion.

LiveView (and similar websocket based implementation such as LiveWire) don't require the Backend API to function and it's actually one of the attractions of this programming paradigm. To be included in the set of Realworld is it mandatory to implement the Backend API, even if it now used in the implementation @geromegrignon ?

**Comment by geromegrignon** [comment]
Thanks for this precious feedback @martinstannard @benregn @mattneel 
RealWorld is so wide now, making it difficult to get the right vision about the details added by a new implementation.
Based on your explanations, I agree with adding another implementation showcasing LiveView.

@fegaeze I'm sorry for not having found out those details, not being familiar with Phoenix myself.
Feel free to contribute and collaborate with anyone willing to help here.

@martinstannard Is LiveView a fullstack solution ?
If so that's fine as a fullstack implementation is meant as a standalone solution rather than just a front and a back implementation working together. So the internals doesn't matter as long as the application provides the same experience for an end user.

**Reply by mattneel** [reply]
I can take a look at putting something together when Phoenix 1.7 comes out of RC status, but yes, it would be fullstack if we kept everything to LiveView.

**Reply by twillzy** [reply]
Hey y'all. My team has put together a full stack implementation with LiveView and [Ash](https://ash-hq.org/). See [repo](https://github.com/team-alembic/realworld).

We have also created a submission on CodebaseShow and it's now pending acceptance. 

@nickbosch @martinstannard
